server/src/main/java/org/opensearch/plugins/PluginsService.java

```diff
@@ -125,12 +125,13 @@ public List<String> getPluginSettingsFilter() {
      * @param pluginsDirectory The directory plugins exist in, or null if plugins should not be loaded from the filesystem
      * @param classpathPlugins Plugins that exist in the classpath which should be loaded
      */
+    @SuppressWarnings("unchecked")
     public PluginsService(
```

Comment:
I think we could keep both constructors, just delegate one to another, wdyt?

Reply:
There's only one non-test usage of this constructor used in [Node.java](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/node/Node.java#L521-L527) and ultimately that's the usage that needs to be updated because tests generally use the [MockNode](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/node/MockNode.java#L87) which extends the Node class.

Reply:
Added a second constructor back in for testing that creates the PluginInfos given a list of classes corresponding to classpathPlugins.